tools/rapids-github-run-id

```diff
@@ -22,61 +23,71 @@ export RAPIDS_SCRIPT_NAME="rapids-github-run-id"
-          --branch "${RAPIDS_REF_NAME}"                       \
-          --commit "${RAPIDS_SHA}"                            \
-          --workflow "${RAPIDS_BUILD_WORKFLOW_NAME:-pr.yaml}" \
-          --json 'createdAt,databaseId'                       \
-          --jq 'sort_by(.createdAt) | reverse | .[0] | .databaseId')}
-    ;;
-  branch)
-    run_id=${GITHUB_RUN_ID:-$(
-      RAPIDS_RETRY_SLEEP=120                                     \
-        rapids-retry --quiet gh run list                         \
-          --repo "${RAPIDS_REPOSITORY}"                          \
-          --branch "${RAPIDS_REF_NAME}"                          \
-          --commit "${RAPIDS_SHA}"                               \
-          --workflow "${RAPIDS_BUILD_WORKFLOW_NAME:-build.yaml}" \
-          --event "push"                                         \
-          --json 'createdAt,databaseId'                          \
-          --jq 'sort_by(.createdAt) | reverse | .[0] | .databaseId')}
-    ;;
-  nightly)
-    # Notice that for nightly runs, this script intentionally does not return GITHUB_RUN_ID.
-    #
-    # In RAPIDS CI, projects often have 2 separate workflows for nightlies:
-    #
-    #   * build.yaml = builds and uploads packages
-    #   * test.yaml  = installs nightly packages and runs tests
-    #
-    # From GitHub's perspective, those are different "runs", with different IDs.
-    # GitHub Actions organizes artifacts by run ID.
-    run_id=$(
-      RAPIDS_RETRY_SLEEP=120                                     \
-        rapids-retry --quiet gh run list                         \
-          --repo "${RAPIDS_REPOSITORY}"                          \
-          --branch "${RAPIDS_REF_NAME}"                          \
-          --commit "${RAPIDS_SHA}"                               \
-          --workflow "${RAPIDS_BUILD_WORKFLOW_NAME:-build.yaml}" \
-          --event "workflow_dispatch"                            \
-          --json 'createdAt,databaseId'                          \
-          --jq 'sort_by(.createdAt) | reverse | .[0] | .databaseId'
-    )
-    ;;
-  *)
-    rapids-echo-stderr "RAPIDS_BUILD_TYPE must be one of [branch, nightly, pull-request]"
-    exit 1
-    ;;
-esac
+# if RAPIDS_BUILD_WORKFLOW_NAME is set, use that instead of any RAPIDS conventions
+if [[ -n "${RAPIDS_BUILD_WORKFLOW_NAME:-}" ]]; then
+  workflow_that_produced_artifacts="${RAPIDS_BUILD_WORKFLOW_NAME}"
+  workflow_that_produced_artifacts_regex="\.github/workflows/${workflow_that_produced_artifacts}"
+else
+  # otherwise, rely on the RAPIDS workflow name conventions
+  case "${RAPIDS_BUILD_TYPE}" in
+    pull-request)
+      workflow_that_produced_artifacts="pr.yaml"
+      workflow_that_produced_artifacts_regex="\\.github/workflows/pr\\.yaml"
+      ;;
+    branch|nightly)
+      workflow_that_produced_artifacts="build.yaml"
+      workflow_that_produced_artifacts_regex="\\.github/workflows/build\\.yaml"
```

Comment:
```suggestion
if [[ -n "${RAPIDS_BUILD_WORKFLOW_NAME:-}" ]]; then
  workflow_that_produced_artifacts="${RAPIDS_BUILD_WORKFLOW_NAME}"
  workflow_that_produced_artifacts_regex="\\.github/workflows/${workflow_that_produced_artifacts}"
else
  # otherwise, rely on the RAPIDS workflow name conventions
  case "${RAPIDS_BUILD_TYPE}" in
    pull-request)
      workflow_that_produced_artifacts="pr.yaml"
      workflow_that_produced_artifacts_regex="\\.github/workflows/pr\\.yaml"
      ;;
    branch|nightly)
      workflow_that_produced_artifacts="build.yaml"
      workflow_that_produced_artifacts_regex="\\.github/workflows/build\\.yaml"
```